### PR TITLE
Fix running blazium on blocked GPUs.

### PR DESCRIPTION
**Pull request by WhalesState**
Patch from godot 4.4.

### CODE REVIEW

**Comment by jss2a98aj** [comment]
If this was cherry-picked, it does not appear that contributor information was kept intact.

File: platform/android/display_server_android.cpp [hunk 1]
```diff
@@ -607,13 +601,26 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
...
-			return;
+#if defined(GLES3_ENABLED)
+			bool fallback_to_opengl3 = GLOBAL_GET("rendering/rendering_device/fallback_to_opengl3");
+			if (fallback_to_opengl3 && rendering_driver != "opengl3") {
+				WARN_PRINT("Your device seem not to support Vulkan, switching to OpenGL 3.");
+				rendering_driver = "opengl3";
+				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
+				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
+			} else
+#endif
```
Comment (Bioblaze) [inline platform/android/display_server_android.cpp:614]: is this correct? or shouldn't this be moved down?

File: platform/ios/display_server_ios.mm [hunk 2]
```diff
@@ -88,13 +88,26 @@
...
+#if defined(GLES3_ENABLED)
+			bool fallback_to_opengl3 = GLOBAL_GET("rendering/rendering_device/fallback_to_opengl3");
+			if (fallback_to_opengl3 && rendering_driver != "opengl3") {
+				WARN_PRINT("Your device seem not to support MoltenVK or Metal, switching to OpenGL 3.");
+				rendering_driver = "opengl3";
+				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
+				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
+			} else
+#endif
+			{
```
Comment (Bioblaze) [inline platform/ios/display_server_ios.mm:102]: same thing here? O_o

File: platform/android/display_server_android.cpp (hunk 1 above)
Comment (WhalesState) [reply platform/android/display_server_android.cpp:614]: Yes, it's correct, the reason why they have done this is to update the rendering method and driver name when a fallback happens, they also have made those two methods public inside the `OS` class.

**Review by WhalesState — COMMENTED** [review]
Reviewed and removed the `RenderingContextDriverMetal` from `display_driver_macos.mm`. 

@jss2a98aj Thank You for helping ^

**Review by jss2a98aj — APPROVED** [review]

**Pull request merged**